src/network/serverpackethandler.cpp

```diff
@@ -1339,10 +1339,37 @@ void Server::handleCommand_RemovedSounds(NetworkPacket* pkt)
 		if (i == m_playing_sounds.end())
 			continue;
 
+		session_t peer_id = pkt->getPeerId();
+
 		ServerPlayingSound &psound = i->second;
-		psound.clients.erase(pkt->getPeerId());
-		if (psound.clients.empty())
+		psound.clients.erase(peer_id);
+			verbosestream << "Server: Sound " << id << " erase peer "
+					<< peer_id << "." << std::endl;
+		if (!psound.can_be_send_later && psound.clients.empty()) {
 			m_playing_sounds.erase(i);
+			verbosestream << "Server: Erase sound " << id << "." << std::endl;
+		}
+		else if (psound.can_be_send_later && !psound.spec.loop) {
+			if (psound.keep_time < m_playing_sounds_time) {
```

Comment:
Why does this exclude looped sounds? Can't a looped sound go out of range just as well, requiring a resend?

Reply:
`keep_time` is not applied to looped sounds.
Looped sounds behavior as `keep_time` is set to infinity.